pkg/controller/stackconfigpolicy/elasticsearch_config_settings.go

```diff
@@ -0,0 +1,149 @@
+				Namespace: es.Namespace,
+			}),
+			Annotations: map[string]string{
+				ElasticsearchConfigAndSecretMountsHashAnnotation: elasticsearchAndMountsConfigHash,
+			},
+		},
+		Data: map[string][]byte{
+			ElasticSearchConfigKey: configDataJSONBytes,
+			SecretsMountKey:        secretMountBytes,
+		},
+	}
+
+	// Set Elasticsearch as the soft owner
+	filesettings.SetSoftOwner(&elasticsearchConfigSecret, policy)
+
+	// Add label to delete secret on deletion of the stack config policy
+	elasticsearchConfigSecret.Labels[commonlabels.StackConfigPolicyOnDeleteLabelName] = commonlabels.OrphanObjectDeleteOnPolicyDelete
+
+	return elasticsearchConfigSecret, nil
+}
+
+func reconcileSecret(ctx context.Context,
+	c k8s.Client,
+	expected corev1.Secret,
+	owner client.Object) error {
+	reconciled := &corev1.Secret{}
+	return reconciler.ReconcileResource(reconciler.Params{
+		Context:    ctx,
+		Client:     c,
+		Owner:      owner,
+		Expected:   &expected,
+		Reconciled: reconciled,
+		NeedsUpdate: func() bool {
+			return !maps.IsSubset(expected.Labels, reconciled.Labels) ||
+				!maps.IsSubset(expected.Annotations, reconciled.Annotations) ||
+				!reflect.DeepEqual(expected.Data, reconciled.Data)
+		},
+		UpdateReconciled: func() {
+			reconciled.Labels = maps.Merge(reconciled.Labels, expected.Labels)
+			reconciled.Annotations = maps.Merge(reconciled.Annotations, expected.Annotations)
+			reconciled.Data = expected.Data
+		},
+	})
+}
+
+// reconcileSecretMounts creates the secrets in SecretMounts to the respective Elasticsearch namespace where they should be mounted to.
+func reconcileSecretMounts(ctx context.Context, c k8s.Client, es esv1.Elasticsearch, policy *policyv1alpha1.StackConfigPolicy) error {
+	for _, secretMount := range policy.Spec.Elasticsearch.SecretMounts {
+		additionalSecret := corev1.Secret{}
+		namespacedName := types.NamespacedName{
+			Name:      secretMount.SecretName,
+			Namespace: policy.Namespace,
+		}
+		if err := c.Get(ctx, namespacedName, &additionalSecret); err != nil {
+			return err
+		}
+
+		// Recreate it in the Elasticsearch namespace, prefix with es name.
+		secretName := esv1.StackConfigAdditionalSecretName(es.Name, secretMount.SecretName)
+		expected := corev1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: es.Namespace,
+				Name:      secretName,
+				Labels: eslabel.NewLabels(types.NamespacedName{
+					Name:      es.Name,
+					Namespace: es.Namespace,
+				}),
+				Annotations: make(map[string]string),
+			},
+			Data: additionalSecret.Data,
+		}
+
+		// Set stackconfigpolicy as a softowner
+		filesettings.SetSoftOwner(&expected, *policy)
+
+		// Set the secret to be deleted when the stack config policy is deleted.
+		expected.Labels[commonlabels.StackConfigPolicyOnDeleteLabelName] = commonlabels.OrphanObjectDeleteOnPolicyDelete
+
+		// Set the original secret created by the user as annotation on the new secret
+		expected.Annotations[SourceSecretAnnotationName] = secretMount.SecretName
```

Comment:
What is the purpose of this annotation?

Reply:
In case we need to debug something, we are able to trace back to the original secret created by the user